src/main/proto/ess.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+option java_multiple_files = true;
+package org.ess;
+
+service LightningStorageService {
+  rpc get(GetRequest) returns (GetResponse);
+  rpc put(PutRequest) returns (PutResponse);
+  //TODO: Add getKeysSummary (Gives version numbers for all keys in a store) and list by prefix API's
+}
+
+message GetRequest {
+  /*
+    All API's operate within a single storeId.
+    StoreId is keyspace identifier. Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+    It is up to clients to use single or multiple store's for their usage.
+   */
+  string storeId = 1;
+  string key = 2;
+}
+message GetResponse {
+  bool success = 1;
+  KeyValue value = 2;
+
+  // Only present when success = false
+  optional ErrorResponse errorResponse = 3;
+}
+
+message PutRequest {
+  string storeId = 1;
+  /*
+  If present, write will only succeed if current DB globalVersion against a storeId is
+  same as in request.
+   */
+  optional int64 globalVersion = 2;
+  repeated KeyValue transactionItems = 3;
+}
+
+message PutResponse {
+  bool success = 1;
+  optional ErrorResponse errorResponse = 2;
+}
+
+message ErrorResponse{
+  ErrorCode errorCode = 1;
+  string message = 2;
+}
+enum ErrorCode{
+  Unknown = 0;
+  ConflictException = 2;
+  InvalidRequestException = 3;
+  InternalServerException = 4;
+}
+
+message KeyValue {
+  string key = 1;
+
+  // Write will only succeed if current DB version against key is same as in request.
+  int64 version = 2;
```

Comment:
I think from client perspective they will have to maintain a current version of object that they have in-memory or secondary store.
It could be more intuitive to start with 0 as first version as it will be same for client and they just call with their current version at-hand instead of any special handling.

Will add it to docs.